src/main/java/org/hiero/sketch/spreadsheet/Bucket.java

```diff
@@ -3,40 +3,35 @@
 /**
  * A Bucket to be used as part of a histogram.
  */
-public class Bucket<T> {
-    private final T leftBoundry;
-    private final T rightBoundry;
+public class Bucket {
+    private final double leftBoundry;
+    private final double  rightBoundry;
     private final boolean leftInclusive;
     private final boolean rightInclusive;
-    private final Comparator<T> greater;
 
-    public Bucket(final T leftBoundry, final boolean leftInclusive,
-                  final T rightBoundry, final boolean rightInclusive, final Comparator<T> greater) {
-        if (greater.compare(leftBoundry, rightBoundry) > 0)
+    public Bucket(final double leftBoundry, final boolean leftInclusive,
+                  final double rightBoundry, final boolean rightInclusive) {
+        if (leftBoundry > rightBoundry)
             throw new IllegalArgumentException("Left boundry cannot be greater than right boundry");
-        else if ((greater.compare(leftBoundry, rightBoundry) == 0 )
-                && !(leftInclusive && rightInclusive))
+        else if ((leftBoundry == rightBoundry) && !(leftInclusive && rightInclusive))
```

Comment:
If leftInclusive == true and rightInclusive == true, then it's okay for leftBoundary to be equal to the rightBoundary? That still is an empty set and should throw an error right?